src/ddmd/expressionsem.d

```diff
@@ -3238,12 +3238,77 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
 
     override void visit(BinAssignExp exp)
     {
+        static if (LOGSEMANTIC)
+        {
+            printf("BinAssignExp::semantic(%s)\n", exp.toChars());
+        }
+
         if (exp.type)
         {
             result = exp;
             return;
         }
 
+        // Binary arithmetic and logic assignment operators for properties
+        // https://issues.dlang.org/show_bug.cgi?id=8006
+        if (exp.e1.op == TOKdotid)
+        {
+            Expression e1 = exp.e1.semantic(sc);
+            if (e1.op == TOKerror)
+            {
+                result = e1;
+                return;
+            }
+
+            Expression e2;
+            switch(exp.op)
+            {
+                case TOKaddass:
+                    e2 = new AddExp(exp.loc, exp.e1, exp.e2);
+                    break;
+                case TOKminass:
+                    e2 = new MinExp(exp.loc, exp.e1, exp.e2);
+                    break;
+                case TOKmulass:
+                    e2 = new MulExp(exp.loc, exp.e1, exp.e2);
+                    break;
+                case TOKdivass:
+                    e2 = new DivExp(exp.loc, exp.e1, exp.e2);
+                    break;
+                case TOKmodass:
+                    e2 = new ModExp(exp.loc, exp.e1, exp.e2);
+                    break;
+                // case TOKpowass:
+                //     See visit(PowAssignExp)
+                case TOKandass:
+                    e2 = new AndExp(exp.loc, exp.e1, exp.e2);
+                    break;
+                case TOKorass:
+                    e2 = new OrExp(exp.loc, exp.e1, exp.e2);
+                    break;
+                case TOKxorass:
+                    e2 = new XorExp(exp.loc, exp.e1, exp.e2);
+                    break;
+                case TOKshlass:
+                    e2 = new ShlExp(exp.loc, exp.e1, exp.e2);
+                    break;
+                case TOKshrass:
+                    e2 = new ShrExp(exp.loc, exp.e1, exp.e2);
+                    break;
+                case TOKushrass:
+                    e2 = new UshrExp(exp.loc, exp.e1, exp.e2);
+                    break;
+                default:
+                    assert(false, "Binary assignment operator was not handled.");
+            }
+
+            if (Expression ex = resolvePropertiesX(sc, e1, e2))
```

Comment:
`resolvePropertiesX` lowers `e1 = e2` into `e1(e2)` if `e1` is a property, so I need `e2` to be `e1 + e2` for that to work.  

Currently DMD errors with `e1() is not an lvalue`, so, if `e1 += e2` is lowered to `e1 = e1 + e2` in the backend, maybe I just need to figure out how to get past that error so the expression makes it to the backend.  But are you sure the backed is doing the lowering?